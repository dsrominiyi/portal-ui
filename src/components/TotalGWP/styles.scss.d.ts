export const totalGwp: string;
export const gwpTotal: string;
export const gwpText: string;
