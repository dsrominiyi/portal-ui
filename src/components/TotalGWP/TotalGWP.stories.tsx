import React from 'react';
import { storiesOf } from '@storybook/react';
import TotalGWP from '.';

import mockData from '../../../__mocks__/mock-data.json';

const stories = storiesOf('TotalGWP', module);

export const render = (policies: Policy[]) => () => <TotalGWP policies={policies} />;

export const storyRenders = {
  withPolicies: render(mockData.policies),
  withNoPolicies: render([])
};

stories.add('With Policies', storyRenders.withPolicies);
stories.add('With No Policies', storyRenders.withNoPolicies);
