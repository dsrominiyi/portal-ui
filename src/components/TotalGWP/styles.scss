@import '../../style/vars';

.total-gwp {
  display: flex;
  flex: 1;
  flex-direction: column;
  justify-content: center;
  color: $brand-charcoal;
}

.gwp-total,
.gwp-text {
  display: flex;
  justify-content: center;
  padding: 5px;
}

.gwp-total {
  font-weight: 500;
  font-size: 2rem;
  color: $brand-success;
}
