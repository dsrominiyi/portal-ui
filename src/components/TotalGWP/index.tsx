import React, { FunctionComponent } from 'react';

import styles from './styles.scss';

interface Props {
  policies: Policy[];
}

export const getTotalGwp = (policies: Policy[]) => {
  const reducer = (accumulator: number, currentValue: number) => accumulator + currentValue;
  const gwp = policies.map(policy => (policy.rate ? policy.rate : 0));

  return gwp.length ? gwp.reduce(reducer) : 0;
};

export const TotalGWP: FunctionComponent<Props> = ({ policies }) => (
  <div className={styles.totalGwp}>
    <div className={styles.gwpTotal}>{`\u00A3${getTotalGwp(policies)}`}</div>
    <div className={styles.gwpText}>Total GWP</div>
  </div>
);

TotalGWP.displayName = 'TotalGWP';
export default TotalGWP;
