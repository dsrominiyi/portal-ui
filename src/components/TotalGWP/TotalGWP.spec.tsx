import { getTotalGwp } from '.';
import { storyRenders } from './TotalGWP.stories';

import testStorySnapshots from '../../helpers/test/testStorySnapshots';

describe('TotalGWP', () => {
  test('component story renders match snapshots', () => {
    testStorySnapshots(storyRenders);
  });

  test('gwp reducer function returns zero with no rate on a policy', () => {
    const policies: Policy[] = [
      {
        id: '1',
        type: 'test',
        insurer: 'test-broker',
        startDate: '2019-01-01',
        endDate: '2020-01-01'
      }
    ];

    expect(getTotalGwp(policies)).toEqual(0);
  });
});
