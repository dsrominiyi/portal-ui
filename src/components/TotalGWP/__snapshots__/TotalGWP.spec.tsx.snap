// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TotalGWP component story renders match snapshots: withNoPolicies 1`] = `
<div
  className="totalGwp"
>
  <div
    className="gwpTotal"
  >
    £0
  </div>
  <div
    className="gwpText"
  >
    Total GWP
  </div>
</div>
`;

exports[`TotalGWP component story renders match snapshots: withPolicies 1`] = `
<div
  className="totalGwp"
>
  <div
    className="gwpTotal"
  >
    £2417.07
  </div>
  <div
    className="gwpText"
  >
    Total GWP
  </div>
</div>
`;
